src/platform/nrfconnect/wifi/ConnectivityManagerImplWiFi.cpp

```diff
@@ -39,124 +39,195 @@ using namespace ::chip::TLV;
+CHIP_ERROR ConnectivityManagerImplWiFi::InitWiFi()
 {
-    return ConnectivityManager::WiFiStationMode::kWiFiStationMode_Disabled;
+    return WiFiManager::Instance().Init();
 }
 
-CHIP_ERROR ConnectivityManagerImplWiFi::_SetWiFiStationMode(ConnectivityManager::WiFiStationMode val)
+ConnectivityManager::WiFiStationMode ConnectivityManagerImplWiFi::_GetWiFiStationMode(void)
 {
+    if (mStationMode != ConnectivityManager::WiFiStationMode::kWiFiStationMode_ApplicationControlled)
+    {
+        mStationMode = (WiFiManager::StationStatus::DISABLED == WiFiManager().Instance().GetStationStatus())
+            ? ConnectivityManager::WiFiStationMode::kWiFiStationMode_Disabled
+            : ConnectivityManager::WiFiStationMode::kWiFiStationMode_Enabled;
+    }
+    return mStationMode;
+}
+
+CHIP_ERROR ConnectivityManagerImplWiFi::_SetWiFiStationMode(ConnectivityManager::WiFiStationMode aMode)
+{
+    VerifyOrReturnError(ConnectivityManager::WiFiStationMode::kWiFiStationMode_NotSupported != aMode,
+                        CHIP_ERROR_UNSUPPORTED_CHIP_FEATURE);
+
+    if (aMode != mStationMode)
+    {
+        mStationMode = aMode;
+        if (mStationMode != ConnectivityManager::WiFiStationMode::kWiFiStationMode_ApplicationControlled)
+        {
+            bool doEnable{ ConnectivityManager::WiFiStationMode::kWiFiStationMode_Enabled == mStationMode };
+            // TODO: when the connection/disconnection callback API is provided
+            // below calls should be used as a base of disconnect callback
+            if (doEnable)
+            {
+                OnStationConnected();
+            }
+            else
+            {
+                OnStationDisconnected();
+            }
+        }
+    }
+
     return CHIP_NO_ERROR;
 }
 
 bool ConnectivityManagerImplWiFi::_IsWiFiStationEnabled(void)
 {
-    return false;
+    return (WiFiManager::StationStatus::DISABLED <= WiFiManager().Instance().GetStationStatus());
 }
 
 bool ConnectivityManagerImplWiFi::_IsWiFiStationApplicationControlled(void)
 {
-    return false;
+    return (ConnectivityManager::WiFiStationMode::kWiFiStationMode_ApplicationControlled == mStationMode);
 }
 
 bool ConnectivityManagerImplWiFi::_IsWiFiStationConnected(void)
 {
-    return false;
+    return (WiFiManager::StationStatus::FULLY_PROVISIONED == WiFiManager().Instance().GetStationStatus());
 }
 
 System::Clock::Timeout ConnectivityManagerImplWiFi::_GetWiFiStationReconnectInterval(void)
 {
-    return System::Clock::kZero;
+    return mWiFiStationReconnectInterval;
 }
 
 CHIP_ERROR ConnectivityManagerImplWiFi::_SetWiFiStationReconnectInterval(System::Clock::Timeout val)
 {
+    mWiFiStationReconnectInterval = val;
     return CHIP_NO_ERROR;
 }
 
 bool ConnectivityManagerImplWiFi::_IsWiFiStationProvisioned(void)
 {
-    return false;
+    return (WiFiManager::StationStatus::CONNECTED <= WiFiManager().Instance().GetStationStatus());
```

Comment:
I think "provisioned" means if the supplicant has been provided SSID&pass (doesn't matter if valid or not). So `CONNECTING` is probably also OK.